src/main/java/org/spongepowered/api/text/TextBuilder.java

```diff
@@ -0,0 +1,1124 @@
+        this.clickAction = clickAction;
+        return this;
+    }
+
+    /**
+     * Returns the current {@link HoverAction} of this builder.
+     *
+     * @return The current hover action or {@link Optional#absent()} if none
+     * @see Text#getHoverAction()
+     */
+    public final Optional<HoverAction<?>> getHoverAction() {
+        return Optional.<HoverAction<?>>fromNullable(this.hoverAction);
+    }
+
+    /**
+     * Sets the {@link HoverAction} that will be executed if the text is hovered
+     * in the chat.
+     *
+     * @param hoverAction The new hover action for the text
+     * @return This text builder
+     * @see Text#getHoverAction()
+     */
+    public TextBuilder onHover(@Nullable HoverAction<?> hoverAction) {
+        this.hoverAction = hoverAction;
+        return this;
+    }
+
+    /**
+     * Returns the current {@link ShiftClickAction} of this builder.
+     *
+     * @return The current shift click action or {@link Optional#absent()} if
+     *         none
+     * @see Text#getShiftClickAction()
+     */
+    public final Optional<ShiftClickAction<?>> getShiftClickAction() {
+        return Optional.<ShiftClickAction<?>>fromNullable(this.shiftClickAction);
+    }
+
+    /**
+     * Sets the {@link ShiftClickAction} that will be executed if the text is
+     * shift-clicked in the chat.
+     *
+     * @param shiftClickAction The new shift click action for the text
+     * @return This text builder
+     * @see Text#getShiftClickAction()
+     */
+    public TextBuilder onShiftClick(@Nullable ShiftClickAction<?> shiftClickAction) {
+        this.shiftClickAction = shiftClickAction;
+        return this;
+    }
+
+    /**
+     * Returns a view of the current children of this builder.
+     *
+     * <p>The returned list is unmodifiable, but not immutable. It will change
+     * if new children get added through this builder.</p>
+     *
+     * @return An unmodifiable list of the current children
+     * @see Text#getChildren()
+     */
+    public final List<Text> getChildren() {
+        return Collections.unmodifiableList(this.children);
+    }
+
+    /**
+     * Appends the specified {@link Text}s to the end of this text.
+     *
+     * @param children The texts to append
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder append(Text... children) {
+        Collections.addAll(this.children, checkNotNull(children, "children"));
+        return this;
+    }
+
+    /**
+     * Appends the specified {@link Text}s to the end of this text.
+     *
+     * @param children The texts to append
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder append(Iterable<? extends Text> children) {
+        Iterables.addAll(this.children, checkNotNull(children, "children"));
+        return this;
+    }
+
+    private static <T> Collection<T> asCollection(Iterable<T> iterable) {
```

Comment:
Only used to prevent a bug in Java 6 that would warn about this without this method.